### ISSUE TITLE
Dashboard won't launch : Exception on /_dash-layout [GET]

### ISSUE THREAD

**Issue by nicolas-lair** [oegedijk/explainerdashboard#167]
I tried to launch a small example to try the explainer dashboard but I get an error from dash when I try to open the dashboard. The command db.run() is ok but I cannot open.

Any idea why I am getting this error ? 

Here is my MWE.

```python
import pandas as pd
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
from explainerdashboard import RegressionExplainer, ExplainerDashboard

#Import the Diabetes Dataset
from sklearn.datasets import load_diabetes

data= load_diabetes()
#print the dataset
data

#create a DataFrame from the dataset
X=pd.DataFrame(data.data,columns=data.feature_names)

#Printing first five rows of the DataFrame
X.head()

y=pd.DataFrame(data.target,columns=["target"])
y.head()

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=1)
print(X_train.shape,y_train.shape,X_test.shape,y_test.shape)

model = RandomForestRegressor(n_estimators=50, max_depth=5)
model.fit(X_train, y_train.values.ravel())

explainer = RegressionExplainer(model, X_test, y_test)#Start the Dashboard
db = ExplainerDashboard(explainer,title="Diabetes Prediction",whatif=False)#Running the app on a local port 3050
db.run()
```
Here is the log after the `db.run()`

```bash
Starting ExplainerDashboard on http://192.168.1.102:8050
Dash is running on http://0.0.0.0:8050/

 * Serving Flask app 'explainerdashboard.dashboards' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://192.168.1.102:8050/ (Press CTRL+C to quit)
```

And the log after I try to access the dashboard at `http://192.168.1.102:8050/`

```bash
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET / HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /assets/bootstrap.min.css?m=1639566730.510363 HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/deps/polyfill@7.v2_0_0m1639566725.12.1.min.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/deps/react@16.v2_0_0m1639566725.14.0.min.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/deps/react-dom@16.v2_0_0m1639566725.14.0.min.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/deps/prop-types@15.v2_0_0m1639566725.7.2.min.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash_bootstrap_components/_components/dash_bootstrap_components.v0_13_1m1639566729.min.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/dash-renderer/build/dash_renderer.v2_0_0m1639566725.min.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/dcc/dash_core_components.v2_0_0m1639566725.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/dcc/dash_core_components-shared.v2_0_0m1639566725.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/html/dash_html_components.v2_0_0m1639566725.min.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:03] "GET /_dash-component-suites/dash/dash_table/bundle.v5_0_0m1639566725.js HTTP/1.1" 200 -
192.168.1.102 - - [15/Dec/2021 12:41:04] "GET /_dash-dependencies HTTP/1.1" 200 -
Exception on /_dash-layout [GET]
Traceback (most recent call last):
  File "C:\Users\nlair\miniconda3\envs\explainer\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\nlair\miniconda3\envs\explainer\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\nlair\miniconda3\envs\explainer\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\nlair\miniconda3\envs\explainer\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\nlair\miniconda3\envs\explainer\lib\site-packages\dash\dash.py", line 560, in serve_layout
    to_json(layout),
  File "C:\Users\nlair\miniconda3\envs\explainer\lib\site-packages\dash\_utils.py", line 21, in to_json
    return to_json_plotly(value)
  File "C:\Users\nlair\miniconda3\envs\explainer\lib\site-packages\plotly\io\_json.py", line 147, in to_json_plotly
    return orjson.dumps(cleaned, option=opts).decode("utf8")
TypeError: Dict key must be str
192.168.1.102 - - [15/Dec/2021 12:41:04] "GET /_dash-layout HTTP/1.1" 500 -
192.168.1.102 - - [15/Dec/2021 12:41:04] "GET /assets/favicon.ico?m=1639566730.5136824 HTTP/1.1" 200 -
```

**Comment by oegedijk** [comment]
Hi @nicolas-lair,

Sorry for late reply, was in the middle of moving house :)

In any case, I'm not able to generate the same error. The code fragment supplied works fine for me and launches the dashboard correctly.

Seems something is going wrong in the internals of dash though. `orjson` is a fast json formatter that speeds up json internals (`dash` sends the result of callback updates, e.g. a new graph, as json packets to the client), however it is also optional in dash. So in case the fault lies with `orjson` you can try uninstalling it: `pip uninstall orjson`.

Does that help?

**Comment by oegedijk** [comment]
Seems like this is indeed an `orjson` issue: https://github.com/plotly/plotly.py/issues/3380

maybe I will remove it from requirements (given that it is optional anyway)...

**Comment by oegedijk** [comment]
Hi @nicolas-lair did uninstalling orjson work for you?

Closing this issue for now...

**Comment by nicolas-lair** [comment]
Hi @oegedijk , thanks for your replies ! Sorry I could not try it yet. I will do and let you know.

**Comment by oegedijk** [comment]
hmmm, at this point I'm wondering whether to just drop orjson as a dependency. It's a nice speedup in dashboard responsiveness, but if too many users have weirdly formed data in their dataframes (or weird from the perspective of orjson), then they will get these errors.

**Comment by VaCH2** [comment]
For now, I only used it for demo purposes with the iris dataset. Running the code locally it worked fine but in my docker setup it broke. As of now, I added an uninstall step in my Dockerfile to make it work, but for my use case making it optional would be the cleanest solution:) But I understand the tradeoff. I will track this issue in case something changes!

**Comment by zoraizelya** [comment]
Mine isn't working yet, even i dont have orjson ... what should I do?

**Comment by zoraizelya** [comment]
Starting ExplainerDashboard on http://192.168.0.108:8050
Dash is running on http://0.0.0.0:8050/

 * Serving Flask app 'explainerdashboard.dashboards'
 * Debug mode: off
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8050
 * Running on http://192.168.0.108:8050
Press CTRL+C to quit

**Comment by oegedijk** [comment]
@zoraizelya 

have you tried uninstalling `pip uninstall orjson`?

Do you have other errors you can report?

**Comment by dkloving** [comment]
I have been experiencing the same issue as @zoraizelya using the Brave browser. However, when I tried with Firefox it loads the dashboard as expected. The strange thing is I have an older dashboard from some years ago running on AWS and Brave browser loads it without issue. I used `pip freeze` to recreate that runtime environment, but this did not solve the problem. The only thing I can think of is that some external / 3rd party js files or something have been updated and the ones for the working app are cached older versions.
